.pre-commit-config.yaml

```diff
@@ -3,52 +3,42 @@ repos:
     rev: 20.8b1
     hooks:
       - id: black
-        args: ['--quiet']
+        args: ["--quiet"]
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.2.0
+    rev: v3.4.0
     hooks:
       - id: check-case-conflict
       - id: end-of-file-fixer
       - id: trailing-whitespace
       - id: check-ast
       - id: check-json
       - id: check-merge-conflict
-      - id: detect-private-key
-      - id: double-quote-string-fixer
   - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.8.3
+    rev: 3.8.4
     hooks:
       - id: flake8
-        additional_dependencies: [
-            'flake8-bugbear==20.1.4',  # Looks for likely bugs and design problems
-            'flake8-comprehensions==3.2.3',  # Looks for unnecessary generator functions that can be converted to list comprehensions
-            'flake8-deprecated==1.3',  # Looks for method deprecations
-            'flake8-use-fstring==1.1',  # Enforces use of f-strings over .format and %s
-            'flake8-print==3.1.4',  # Checks for print statements
-            'flake8-docstrings==1.5.0',  # Verifies that all functions/methods have docstrings
-            'flake8-type-annotations==0.1.0',  # Looks for misconfigured type annotations
-        ]
```

Comment:
Why did you remove these?

Reply:
for the typing stuff - you already had mypy in place, no need to have flake8 do an inferior job. For the rest of it - these did not actually contribute to find any errors in the code base. If we want these type of checks we better use pylint (although its somewhat dubious as well):

Reply:
also the main reason - to enable running flake8 against the test files as well. This allowed me to find a lot of issues and fix them (especially missing and imports and unused variables etc). 

> Looks like there's almost 4000 changes here, where the strong majority is just single-to-double quotes. Please revert the quote-changes.

not possible, this is black.

Reply:
> not possible, this is black.

No, this happened because you removed the `double-quote-string-fixer` from the pre-commit config and ran it on the project.

> for the typing stuff - you already had mypy in place, no need to have flake8 do an inferior job

What does mypy have to do with:

- Fixing quotes
- Detecting private keys (I don't really care about this hook)
- Detecting bad list comprehensions
- Bugbear
- Method deprecations 
- Checking for print statements
- Checking for docstrings
- Checking that f-strings are used over format

The only flake8-hook related to mypy is the type annotations one, and mypy does not enforce type annotations. For example, this code would pass mypy's tests without errors

```python
def test(x, y):
    return x + y
    
test(1, 3)
```

With the flake8 type-annotations hook this would not pass the pre-commit check.

-------------------------

I think the best thing would be to undo these removals and run `pre-commit run --all-files`. That should eliminate the excess changes.

Let me know if you want me to do it :)

Reply:
followed your suggestion. it worked!